overlord/ifacestate/helpers.go

```diff
@@ -310,6 +310,26 @@ func (m *InterfaceManager) reloadConnections(snapName string) ([]string, error)
 	return result, nil
 }
 
+func (m *InterfaceManager) setupPhasedSecurity(task *state.Task, snaps []*snap.Info, opts []interfaces.ConfinementOptions) error {
+	st := task.State()
+
+	// Setup all affected snaps, start with the most important security
+	// backend and run it for all snaps. See LP: 1802581
+	for _, backend := range m.repo.Backends() {
+		for i, snapInfo := range snaps {
+			st.Unlock()
```

Comment:
Can we just `st.Unlock()` and `defer st.Lock()` before the loop (I know this is how the old `setupSnapSecurity` does that, just curious.

Reply:
the reason is done like that is that the systemd backend can take a long time